File: languages/java/app-encryption/src/test/java/com/godaddy/asherah/appencryption/AppEncryptionJsonImplTest.java
package com.godaddy.asherah.appencryption;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

import org.json.JSONObject;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import com.godaddy.asherah.appencryption.envelope.EnvelopeEncryption;
import com.godaddy.asherah.appencryption.utils.Json;

@ExtendWith(MockitoExtension.class)
class AppEncryptionJsonImplTest {

  @Mock
  EnvelopeEncryption<String> envelopeEncryption;
  @InjectMocks
  AppEncryptionJsonImpl<String> appEncryptionJsonImpl;
  
  @Test
  void testConstructor() {
    AppEncryption<?, ?> appEncryption = new AppEncryptionJsonImpl<>(envelopeEncryption);
    assertNotNull(appEncryption);
  }

  @Test
  void testDecrypt() {
    JSONObject expectedJson = new JSONObject("{\"some_key\": 123}");
    byte[] utf8Bytes = new Json(expectedJson).toUtf8();
    when(envelopeEncryption.decryptDataRowRecord(any())).thenReturn(utf8Bytes);

    JSONObject actualJson = appEncryptionJsonImpl.decrypt("some data row record");
    assertTrue(expectedJson.similar(actualJson));
  }

  @Test
  void testEncrypt() {
    String expectedDataRowRecord = "some data row record";
    when(envelopeEncryption.encryptPayload(any())).thenReturn(expectedDataRowRecord);

    String actualDataRowRecord = appEncryptionJsonImpl.encrypt(new JSONObject("{\"some_key\": 123}"));
    assertEquals(expectedDataRowRecord, actualDataRowRecord);
  }

  @Test
  void testCloseSuccess() {
    appEncryptionJsonImpl.close();
    
    // Verify proper resources are closed
    verify(envelopeEncryption).close();
  }

  @Test
  void testCloseWithCloseFailShouldReturn() {
    doThrow(RuntimeException.class).when(envelopeEncryption).close();
    appEncryptionJsonImpl.close();
    
    verify(envelopeEncryption).close();
  }

}

File: languages/csharp/AppEncryption/AppEncryption.Tests/AppEncryption/AppEncryptionJsonImplTest.cs
using System;
using GoDaddy.Asherah.AppEncryption.Envelope;
using Moq;
using Newtonsoft.Json.Linq;
using Xunit;

namespace GoDaddy.Asherah.AppEncryption.Tests.AppEncryption
{
    [Collection("Logger Fixture collection")]
    public class AppEncryptionJsonImplTest
    {
        private readonly Mock<IEnvelopeEncryption<string>> envelopeEncryptionMock;
        private readonly AppEncryptionJsonImpl<string> appEncryptionJsonImpl;

        public AppEncryptionJsonImplTest()
        {
            envelopeEncryptionMock = new Mock<IEnvelopeEncryption<string>>();
            appEncryptionJsonImpl = new AppEncryptionJsonImpl<string>(envelopeEncryptionMock.Object);
        }

        [Fact]
        private void TestConstructor()
        {
            AppEncryptionJsonImpl<string> appEncryption = new AppEncryptionJsonImpl<string>(envelopeEncryptionMock.Object);
            Assert.NotNull(appEncryption);
        }

        [Fact]
        private void TestDecrypt()
        {
            const string json = @"{key:'some_key', value:123}";
            JObject expectedJson = JObject.Parse(json);
            byte[] utf8Bytes = new Asherah.AppEncryption.Util.Json(expectedJson).ToUtf8();

            envelopeEncryptionMock.Setup(x => x.DecryptDataRowRecord(It.IsAny<string>())).Returns(utf8Bytes);

            JObject actualJson = appEncryptionJsonImpl.Decrypt("some data row record");
            Assert.Equal(expectedJson,  actualJson);
        }

        [Fact]
        private void TestEncrypt()
        {
            const string expectedDataRowRecord = "some data row record";
            const string json = @"{key:'some_key', value:123}";
            JObject jObject = JObject.Parse(json);

            envelopeEncryptionMock.Setup(x => x.EncryptPayload(It.IsAny<byte[]>())).Returns(expectedDataRowRecord);

            string actualDataRowRecord = appEncryptionJsonImpl.Encrypt(jObject);
            Assert.Equal(expectedDataRowRecord, actualDataRowRecord);
        }

        [Fact]
        private void TestDispose()
        {
            appEncryptionJsonImpl.Dispose();

            // Verify proper resources are closed
            envelopeEncryptionMock.Verify(x => x.Dispose());
        }

        [Fact]
        private void TestCloseWithCloseFailShouldReturn()
        {
            envelopeEncryptionMock.Setup(x => x.Dispose()).Throws<Exception>();
            appEncryptionJsonImpl.Dispose();
            envelopeEncryptionMock.Verify(x => x.Dispose());
        }
    }
}
